### PR TITLE
pythonPackages.quart: init at 0.17.0

### PR DESCRIPTION
**Pull request by dgliwka**
###### Description of changes

[Quart](https://pgjones.gitlab.io/quart/) - an async Python web microframework.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/quart/default.nix [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+  propagatedBuildInputs = [
+    click
+    aiofiles
+    jinja2
+    hypercorn
+    itsdangerous
+    blinker
+    werkzeug
+    poetry
```
Comment (pbsds) [inline pkgs/development/python-modules/quart/default.nix:45]: Could this be substituted with `poetry-core`?

File: pkgs/development/python-modules/quart/default.nix [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    hypothesis
+    mock
+    pytest-asyncio
+  ];
+
+  pythonImportsCheck = [ "quart" ];
+
+  meta = with lib; {
+    homepage = "https://pgjones.gitlab.io/quart/";
+    description = "Python ASGI web microframework";
```
Comment (pbsds) [inline pkgs/development/python-modules/quart/default.nix:59]: This is likely more searchable:

"ASGI web microframework, built as an asyncio alternative to Flask"

**Comment by pbsds** [comment]
Result of `nixpkgs-review pr 168268` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.quart</li>
    <li>python39Packages.quart</li>
  </ul>
</details>

File: pkgs/development/python-modules/quart/default.nix (hunk 2 above)
Comment (dgliwka) [reply pkgs/development/python-modules/quart/default.nix:59]: I have changed it as you suggested. Thanks.

File: pkgs/development/python-modules/quart/default.nix (hunk 1 above)
Comment (dgliwka) [reply pkgs/development/python-modules/quart/default.nix]: I changed `poetry` to `poetry-core` but `itsdangerous`, `blinker` and `werkzeug` had to stay to build. Thanks for taking a look :-).

File: pkgs/development/python-modules/quart/default.nix [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    werkzeug
+    hypothesis
+    itsdangerous
+    pytest-asyncio
+  ];
+
+  pythonImportsCheck = [ "quart" ];
+
+  meta = with lib; {
+    homepage = "https://pgjones.gitlab.io/quart/";
```
Comment (pbsds) [inline pkgs/development/python-modules/quart/default.nix:56]: Seems like upstream just moved: https://github.com/pallets/quart

File: pkgs/development/python-modules/quart/default.nix [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+}:
+
+buildPythonPackage rec {
+  pname = "quart";
+  version = "0.17.0";
+  disabled = pythonOlder "3.7";
+  format = "pyproject";
+
+  src = fetchFromGitLab {
+    owner = "pgjones";
```
Comment (ozkutuk) [inline pkgs/development/python-modules/quart/default.nix:25]: Quart is now a Pallets project, hence the repo moved to https://github.com/pallets/quart

File: pkgs/development/python-modules/quart/default.nix (hunk 3 above)
Comment (ozkutuk) [reply pkgs/development/python-modules/quart/default.nix:56]: As @pbsds mentioned, we can move this PR to 0.18.0. It requires `Werkzeug >= 2.2.0`, but that is [already merged to staging](https://github.com/NixOS/nixpkgs/pull/187000), so I think it shouldn't be an issue if we make this PR target the `staging` branch.

File: pkgs/development/python-modules/quart/default.nix (hunk 4 above)
Comment (dgliwka) [reply pkgs/development/python-modules/quart/default.nix]: Changed source to `github.com/pallets/quart`.
Bumped up version to `0.18.0`.
Changed PR target to `staging`

Thanks :-)

File: pkgs/development/python-modules/quart/default.nix [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    substituteInPlace pyproject.toml \
+        --replace "--no-cov-on-fail" ""
+  '';
+
+  propagatedBuildInputs = [
+    click
+    aiofiles
+    jinja2
+    hypercorn
+  ];
```
[the remark this answers is not in the document]
Comment (onny) [reply pkgs/development/python-modules/quart/default.nix]: Is this sorted out?

[the remark this answers is not in the document]
Comment (pbsds) [reply pkgs/development/python-modules/quart/default.nix]: Eh, the import check passes, it should be fine.

**Comment by onny** [comment]
@ofborg build pythonPackages.quart
@ofborg build python3Packages.quart

**Comment by oxzi** [comment]
Thanks a lot for this PR which came handy for me. Due to the time it lay around, there were some small upstream changes which I reflected locally. Please find a patch below and feel free to apply it on top of your branch. It works fine for me after rebasing your PR's branch to the current master.


```
From ca0f1e9352d3840c9417eb0fda6f5ad4d841d232 Mon Sep 17 00:00:00 2001
From: Alvar Penning <post@0x21.biz>
Date: Thu, 29 Dec 2022 00:06:57 +0100
Subject: [PATCH] pythonPackages.quart: 0.18.0 -> 0.18.3

Next to the version bump, the order of the used python modules was
sorted alphabetically as they are present in the pyproject.toml file.
The description now also reflects the upstream's description.

https://github.com/pallets/quart/releases/tag/0.18.3
---
 .../python-modules/quart/default.nix          | 54 ++++++++++++-------
 1 file changed, 36 insertions(+), 18 deletions(-)

diff --git a/pkgs/development/python-modules/quart/default.nix b/pkgs/development/python-modules/quart/default.nix
index 934abbe5397..ed7e384a6e7 100644
--- a/pkgs/development/python-modules/quart/default.nix
+++ b/pkgs/development/python-modules/quart/default.nix
@@ -1,23 +1,30 @@
 { lib
-, buildPythonPackage
 , fetchFromGitHub
+, buildPythonPackage
 , pythonOlder
-, click
+
 , aiofiles
-, werkzeug
-, jinja2
-, hypercorn
-, itsdangerous
 , blinker
+, click
+, hypercorn
 , hypothesis
-, pytestCheckHook
-, pytest-asyncio
+, importlib-metadata
+, itsdangerous
+, jinja2
+, markupsafe
+, mock
 , poetry-core
+, pydata-sphinx-theme
+, pytest-asyncio
+, pytestCheckHook
+, python-dotenv
+, typing-extensions
+, werkzeug
 }:

 buildPythonPackage rec {
   pname = "quart";
-  version = "0.18.0";
+  version = "0.18.3";
   disabled = pythonOlder "3.7";
   format = "pyproject";

@@ -25,7 +32,7 @@ buildPythonPackage rec {
     owner = "pallets";
     repo = pname;
     rev = version;
-    sha256 = "sha256-xpMvCOFoSpuit+KnLs3iWwneK++mLgjD7E6vTUUMwaU=";
+    sha256 = "sha256-aQM8kEhienBG+/zQQ8C/DKiDIMF3l9rq8HSAvg7wvLM=";
   };

   prePatch = ''
@@ -33,25 +40,36 @@ buildPythonPackage rec {
         --replace "--no-cov-on-fail" ""
   '';

-  propagatedBuildInputs = [ click aiofiles jinja2 hypercorn ];
+  propagatedBuildInputs = [
+    aiofiles
+    blinker
+    click
+    hypercorn
+    importlib-metadata
+    itsdangerous
+    jinja2
+    markupsafe
+    pydata-sphinx-theme
+    python-dotenv
+    typing-extensions
+    werkzeug
+  ];

   checkInputs = [
-    pytestCheckHook
     poetry-core
-    blinker
-    werkzeug
+
     hypothesis
-    itsdangerous
+    mock
     pytest-asyncio
+    pytestCheckHook
   ];

   pythonImportsCheck = [ "quart" ];

   meta = with lib; {
     homepage = "https://quart.palletsprojects.com/en/latest/";
-    description =
-      "ASGI web microframework, built as an asyncio alternative to Flask";
+    description = "A Python ASGI web microframework with the same API as Flask";
     license = licenses.mit;
-    maintainers = with maintainers; [ dgliwka ];
+    maintainers = with maintainers; [ dgliwka oxzi ];
   };
 }
--
2.38.1
```

**Comment by mweinelt** [comment]
Superseded by 14c9c309823b248aa352d92ed519823c01a1d4de

**Pull request closed without merging**